File: 백준/Silver/1966. 프린터 큐/프린터 큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        int testCase = Integer.parseInt(bf.readLine());

        for (int i = 0; i < testCase; i++) {

            StringTokenizer nAndTarget = new StringTokenizer(bf.readLine());
            int n = Integer.parseInt(nAndTarget.nextToken());
            int target = Integer.parseInt(nAndTarget.nextToken());

            StringTokenizer strNums = new StringTokenizer(bf.readLine());
            Queue<int[]> que = new LinkedList<>();
            Integer[] temp = new Integer[n];
            int idx = 0;
            while (strNums.hasMoreTokens()) {
                int num = Integer.parseInt(strNums.nextToken());
                que.add(new int[]{idx, num});
                temp[idx] = num;
                idx++;
            }
            Arrays.sort(temp, Collections.reverseOrder());
            int seq = 0;

            loop1:
            for (Integer num : temp) {
                seq++;
                while (!que.isEmpty()) {
                    int[] q = que.remove();
                    if (q[1] == num) {
                        if (q[0] == target) {
                            System.out.println(seq);
                            break loop1;
                        }
                        break;
                    }
                    else {
                        que.add(q);
                    }
                }

            }

        }
        bf.close();
    }
}

File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
import sys

t = int(input())
for i in range(t):
    n, m = map(int,sys.stdin.readline().rstrip().split())
    nums = list(map(int,sys.stdin.readline().rstrip().split()))
    temp = list()
    for idx in range(len(nums)):
        temp.append(nums[idx])
        nums[idx] = [idx,nums[idx]]

    temp.sort(reverse=True)

    seq = 0
    for num in temp :
        seq += 1
        isEnd = False
        while True :
            if nums[0][1] == num :
                if nums[0][0] == m :
                    print(seq)
                    isEnd = True
                    break
                nums.pop(0)
                break
            nums.append(nums.pop(0))

        if isEnd :
            break